core/trino-main/src/main/java/io/trino/connector/system/GlobalSystemConnector.java

```diff
@@ -93,11 +95,12 @@ public ConnectorSplitManager getSplitManager()
             @Override
             public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorTableFunctionHandle functionHandle)
             {
-                if (functionHandle instanceof SequenceFunctionHandle sequenceFunctionHandle) {
-                    return getSequenceFunctionSplitSource(sequenceFunctionHandle);
-                }
-
-                throw new UnsupportedOperationException();
+                return tableFunctionFeaturesFactories.stream()
+                        .map(factory -> factory.getSplitSource(functionHandle))
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .findFirst()
```

Comment:
What if the are many matches? I think you shoulld use `toOptional` collector.

Reply:
check size added, we should have only zero or one